app/src/androidTest/java/org/mozilla/focus/activity/URLCompletionTest.java

```diff
@@ -67,4 +92,192 @@ public void CompletionTest() throws InterruptedException, UiObjectNotFoundExcept
         TestHelper.hint.waitForExists(waitingTime);
         assertTrue (TestHelper.inlineAutocompleteEditText.getText().equals("http://www.mozilla.org"));
     }
+
+    @Test
+    public void CustomCompletionTest() throws UiObjectNotFoundException {
+        OpenCustomCompleteDialog();
+
+        // Enable URL autocomplete, and add URL
+        toggleAutocomplete();
+        addAutoComplete(site);
+        exitToTop();
+
+        // Check for custom autocompletion
+        checkACOn(site);
+        // Remove custom autocompletion site
+        OpenCustomCompleteDialog();
+        removeACSite();
+        exitToTop();
+
+        // Check autocompletion
+        checkACOff(site.substring(0,3));
+
+        // Reset to disabled
+        OpenCustomCompleteDialog();
+        toggleAutocomplete();       // Disable autocomplete
+    }
+
+    @Test
+    // add custom autocompletion site, but disable autocomplete
+    public void DisableCCwithSiteTest() throws UiObjectNotFoundException {
+        OpenCustomCompleteDialog();
+        toggleAutocomplete();
+        addAutoComplete(site);
+        Espresso.pressBack();
+        toggleAutocomplete();       // Disable autocomplete
+        Espresso.pressBack();
+        Espresso.pressBack();
+        checkACOff(site.substring(0,3));
+
+        // Now enable autocomplete
+        OpenCustomCompleteDialog();
+        toggleAutocomplete();
+        Espresso.pressBack();
+        Espresso.pressBack();
+
+        // Check autocompletion
+        checkACOn(site);
+        // Cleanup
+        OpenCustomCompleteDialog();
+        removeACSite();
+        Espresso.pressBack();
+        toggleAutocomplete();       
+    }
+
+    @Test
+    public void DuplicateACSiteTest() {
```

Comment:
\[PMD\] INFO: The method 'DuplicateACSiteTest\(\)' has a NCSS line count of 14\.